packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -52,10 +51,45 @@ const toJsonBKey = <T extends keyof NestedFilterQueryFields>(nestedField: T): Js
   return [field, value].join(JSONB_FIELD_DELIMITER) as JsonBKey<T>;
 };
 
+// Inspired by Google Analytics filter: https://developers.google.com/analytics/devguides/reporting/core/v3/reference?hl=en#filters
+const operatorToSqlComparator = {
+  '==': null, // Exact match
+  '!=': '!=', // Does not match
+  '=@': 'ilike', // Contains sub string
+};
+const filterOperators = Object.keys(operatorToSqlComparator);
+
+const formatComparisonValue = (value: string | string[], operator: string) => {
+  if (operator === '=@' && typeof value === 'string') {
+    return `%${value}%`;
+  }
+
+  return value;
+};
+
+const convertValueToAdvancedCriteria = (operator: string, value: string | string[]) => {
```

Comment:
```suggestion
const convertValueToAdvancedCriteria = (operator: Operator, value: string | string[]) => {
```
This one is important, as otherwise the lookup in the `operatorToSqlComparator` won't give us a typesafe response